Implement DAL and DB structure for Game entity

Oleksii-Stepanyk opened the pull request:
### This PR implements the general structure for DAL and `Game` entity-related classes/interfaces inside DAL.

### User stories:
1. Add `Game` logic for DAL
2. Configure `Game` table in DB

### Changes made:
 - Created generic `IRepository` and `IGameRepository` interface
 - Created `GameRepository` class with implementation
 - Created `Game` entity
 - Created `OnlineGameStoreDbContext` with `DbSet<Game>` field
 - Created `ServiceCollectionExtensions` class 
 - Created table configuration for `Game` entity
 - Created unit tests for the `GameRepository` class
 - Created generic `RepositoryCreator` and `GameRepositoryCreator` classes for test purposes
 - Generated database migration

### Notes:
1. Tests:
      - Unit tests cover all 5 CRUD methods for `GameRepository` using InMemoryDatabase
2. Interfaces:
      - Since this is the first PR related to DAL, I created a generic `IRepository` interface, which can be used with any entity, and contains only basic CRUDs. The same principle applies with the `RepositoryCreator` class for test purposes.
3. ServiceCollectionExtensions:
      - This class will be used in the UI layer for DI. Instead of manually configuring all services and other things in one place, we can split them up by layer and call only one method per layer.

hermanhavva reviewed (changes requested):
This part is solid and looks great. 
However, there are a few places where you missed `async` and `await`.

hermanhavva reviewed (approved):
LGTM

Oleksii-Stepanyk commented on OnlineGameStore.DAL/Configurations/GameConfiguration.cs:
```diff
@@ -0,0 +1,42 @@
+        builder.Property(g => g.Description)
+            .HasColumnName("description")
+            .IsRequired()
+            .HasMaxLength(4096);
```
Should I add it to DAL, or create a separate project for shared data?

Oleksii-Stepanyk commented on OnlineGameStore.DAL/Repositories/GameRepository.cs:
```diff
@@ -0,0 +1,69 @@
+
+    public Task<IEnumerable<Game>> GetAllAsync()
+    {
+        return Task.FromResult(context.Games.AsEnumerable());
```
This implementation is now outdated
Current is:
```
public async Task<IEnumerable<Game>> GetAllAsync()
    {
        return await context.Games.ToListAsync();
    }
```

Oleksii-Stepanyk commented:
## Update
 - Fixed minor things inside `GameRepository`
 - Added `GameConstants` static class and used where needed
 - Changed type of ids in `Game` entity from `int` to `Guid`
 - License, genre, and platform in the `Game` entity can be null - this is needed in case some entity is deleted
 - Created a new migration with new IDs
